lib/videojs-playlist-ui.js

```diff
@@ -1,25 +1,188 @@
 /*! videojs-playlist-ui - v0.0.0 - 2015-3-12
  * Copyright (c) 2015 Brightcove
  * Licensed under the Apache-2.0 license. */
-(function(window, videojs) {
-  'use strict';
-
-  var defaults = {
-        option: true
-      },
-      playlistUi;
-
-  /**
-   * Initialize the plugin.
-   * @param options (optional) {object} configuration for the plugin
-   */
-  playlistUi = function(options) {
-    var settings = videojs.util.mergeOptions(defaults, options),
-        player = this;
-
-    // TODO: write some amazing plugin code
-  };
-
-  // register the plugin
-  videojs.plugin('playlistUi', playlistUi);
-})(window, window.videojs);
+import * as playlist from 'videojs-playlist';
+
+const defaults = {
+  className: 'vjs-playlist'
+};
+
+const createThumbnail = function(thumbnail) {
+  if (!thumbnail) {
+    let placeholder = document.createElement('div');
+    placeholder.className = 'vjs-playlist-thumbnail';
+    return placeholder;
+  }
+
+  let picture = document.createElement('picture');
+  picture.className = 'vjs-playlist-thumbnail';
+
+  if (typeof thumbnail === 'string') {
+    // simple thumbnails
+    let img = document.createElement('img');
+    img.src = thumbnail;
+    picture.appendChild(img);
+  } else {
+    // responsive thumbnails
+
+    // additional variations of a <picture> are specified as
+    // <source> elements
+    for (let i = 0; i < thumbnail.length - 1; i++) {
```

Comment:
Using for-of pulls in a whole bunch of iterator support so I decided to do things the old fashioned way for compatibility.